mesonbuild/cmake/traceparser.py

```diff
@@ -231,7 +231,13 @@ def parse(self, trace: T.Optional[str] = None) -> None:
 
         for ctgt in self.custom_targets:
             ctgt.outputs = pathlist_gen(ctgt._outputs_str)
+            temp = ctgt.command
             ctgt.command = [strlist_gen(x) for x in ctgt.command]
+            for command, src in zip(ctgt.command, temp):
+                if command[0] == "":
+                    raise CMakeException(
+                        "We evaluated the cmake variable '{}' to an empty string, which is not a valid path to an executable.".format(src[0])
```

Comment:
Maybe:

"The cmake variable '{}' is an empty string, which is not a valid path to an executable."

Reply:
I think its important to stress here that meson made the variable evaluation, not cmake, because it is likely, that a variable is not an empty string if it is evaluated by cmake. i.e. correctly, but it is an empty string it is evaluated by meson i.e. incorrectly.